test/browser_tests/spec_suites/organisms/header.js

```diff
@@ -135,11 +134,31 @@ describe( 'Header', function() {
               expect( _dom.overlay.isDisplayed() ).toBe( true );
             } );
         } );
+
+        describe( 'then click search', function() {
+          it( 'should show the search and hide menu', function() {
+            browser.driver.actions().click( _dom.megaMenuTrigger ).perform();
+            browser.driver.wait( _dom.megaMenuContent.isDisplayed() );
+            browser.driver.actions().click( _dom.globalSearchTrigger ).perform()
+              .then( function() {
+                expect( _dom.globalSearch.isDisplayed() ).toBe( true );
+
+              } );
+          } );
+        } );
+      } );
+
+      describe( 'click search, then click mega menu', function() {
+        it( 'should show the megamenu and hide search', function() {
+          browser.driver.actions().click( _dom.globalSearchTrigger ).perform();
+          browser.driver.wait( _dom.megaMenuContent.isDisplayed() );
```

Comment:
> EDIT: Which makes me wonder if wait is even doing anything or can be removed from all these tests

This is a tricky one because the `wait` line could be removed and the test would still pass because it's checking that the mega menu shows after clicking the trigger, which will always be true (if the menu is working). Ideally it should be testing that the search is displayed, then after clicking the menu trigger, check that the megamenu is displayed and search is no longer displaying.